### PR TITLE
Signup: hide preview in domains step

### PR DESCRIPTION
**Pull request by ramonjd**
## Changes proposed in this Pull Request

Hi GIF lovers!

This PR proposes two changes:

### 1. Toggle the preview from the domains step so that it hides for loading states, search results and trademark notices

For starters, the domain suggestions muscles the preview out of view as soon as you enter a domain. Props to @p-jackson for the idea.

Aug-28-2019 14-59-03

Secondly, when you search for and select a trademarked domain such as `dairyqueen.dev`, you'll see a trademark warning and acknowledgment screen. 

The presence of the site preview on this page is undesirable, but also prevents the user from clicking the floating acknowledgement button, like that [moving button you can never click](http://jsfiddle.net/9CDtE/4/).

Aug-28-2019 12-03-18

### 2. Forcibly relocate the site mockup component from the current step rendering method in order to prevent refresh of the site preview in between steps.

The heading says it all, but please accept this GIF:

Aug-28-2019 15-00-30

## Testing instructions

~To test properly,~ https://github.com/Automattic/wp-calypso/pull/35861 is merged! ~needs to be merged!~

Go through the onboarding flow at `/start`!

Expectations:

- [ ] The site preview should not reload at each step. 
- [ ] There should only be a preview at the domains step when there are no domain suggestion results
- [ ] There should be no preview on any other domain screen, such as the trademark notice, domain transfer or mapping. To test the trademark screen, search for and select `dairyqueen.dev` or `tacobell.page` as your domain, **unsandboxed**.
- [ ] You should be showered with adulation for testing this PR.

### CODE REVIEW

File: client/signup/main.jsx [hunk 1]
```diff
@@ -536,7 +536,6 @@ class Signup extends React.Component {
 								hideFreePlan={ hideFreePlan }
 								{ ...propsFromConfig }
 							/>
-							{ this.props.shouldShowMockups && <SiteMockups stepName={ this.props.stepName } /> }
 						</>
```
Comment (p-jackson) [inline client/signup/main.jsx]: This fragment can be removed now. Just means we can trim off one level of indentation :)

**Comment by ramonjd** [comment]
> I think we do lose some of the continuity between the the different steps by removing it. I was thinking maybe it could hide the preview after the rich domain UI pops up. 

This is a fair point, and I like the compromise you suggested 👍I'll build it in. Cheers!

**Comment by spen** [comment]
> So I'm wondering if when the site preview isn't rendered, do we need a bit of padding, say 24px or so like the site mockup help tip ("Scroll to top" on the other steps)?

I had planned a while ago to look at adding that bottom padding any time Inline Help was present.
For now I think adding it to just this specific component would suffice but It'd be cool if we could bake that in to the layout proper eventually, so that others don't need to worry about it for their new page/component.

**Comment by ramonjd** [comment]
Thanks @spen and @andrewserong for testing this one. 

Added a bottom margin as you suggested.

I've sprayed a few conditions around, so there's probably a neater way to do it.

I'm also now checking for refreshes on step subscreens, e.g., transfer domain. The preview won't show there either.

**Comment by andrewserong** [comment]
This is more of a note to self, but while testing I noticed an unrelated styling issue on mobile on the `domains-launch` step where the width of the page is a bit wider than the viewport on mobile (this issue exists on Master). I'll write up an issue for this separately later on today:

image

**Pull request merged**